### PR TITLE
added enemies, bullet deletion upon hitting them

### PR DESCRIPTION
**Pull request by tjforys**

### CODE REVIEW

File: amongus.py [hunk 1]
```diff
@@ -24,9 +25,13 @@ def main():
 
     amongus = pygame.image.load(FilePaths.png_amogus).convert_alpha()
     amongus = pygame.transform.scale(amongus, (40, 52))
-
+    enemy_sprite = pygame.image.load(r"sprite/enemy.png").convert_alpha()
+    enemy_sprite = pygame.transform.scale(enemy_sprite, (40, 52))
     player = Player(sprite=amongus, position=[250, 250], radius=10, speed=1)
+    enemy = Enemy(sprite= enemy_sprite, pos_x=1000, pos_y=1000, speed=0.5, health=10)
+    enemy_2 = Enemy(sprite= enemy_sprite, pos_x=0, pos_y=0, speed=0.5, health=10)
```
Comment (Xaxedd) [inline amongus.py:32]: put this in enemies list in line 34

File: amongus.py [hunk 2]
```diff
@@ -39,20 +44,28 @@ def main():
...
         if not bullets:
             print("no more boolets")
  
         Draw.draw_player(screen.screen, player)
+        for enemy_obj in enemies:
+            for bullet in reversed(bullets):
+                if enemy_obj.is_hit(bullet):
+                    enemy_obj.take_damage(1)
+            Draw.draw_enemy(screen=screen, enemy=enemy_obj)
+            enemy_obj.move(player.position[0], player.position[1])
```
Comment (Xaxedd) [inline amongus.py:66]: make functions for this

File: amongus.py [hunk 3]
```diff
@@ -39,20 +44,28 @@ def main():
...
+        bullets = list(filter(lambda b: all([not enemy_obj.is_hit(b) for enemy_obj in enemies]), bullets))
+        enemies = list(filter(lambda e: e.health>0, enemies))
```
Comment (Xaxedd) [inline amongus.py:68]: make functions for those instead of lambdas

File: objects/enemy.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+    def is_hit(self, bullet: Bullet):
+        if bullet.position[0] - bullet.radius < self.pos_x +self.hitbox[0] and bullet.position[0] + bullet.radius > self.pos_x:
+            if bullet.position[1] - bullet.radius < self.pos_y +self.hitbox[1] and bullet.position[1] + bullet.radius > self.pos_y: 
+                return True          
+        return False
+    
+
+    def take_damage(self, amount):
```
Comment (Xaxedd) [inline objects/enemy.py:34]: add type annotation to ```amount```

File: amongus.py (hunk 1 above)
Comment (tjforys) [reply amongus.py:32]: do(i)n(b)e(d)

File: amongus.py (hunk 3 above)
Comment (tjforys) [reply amongus.py:68]: DOINBED

File: objects/enemy.py (hunk 4 above)
Comment (tjforys) [reply objects/enemy.py:34]: done

**Pull request merged**